src/documentation/ToolkitDocumentation/TopicItem.tsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * (c) 2021, Micro:bit Educational Foundation and contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+import { Button } from "@chakra-ui/button";
+import { Box, BoxProps, HStack, Stack, Text } from "@chakra-ui/layout";
+import { Portal } from "@chakra-ui/portal";
+import { forwardRef } from "@chakra-ui/system";
+import { Ref, RefObject, useLayoutEffect, useRef, useState } from "react";
+import { useSplitViewContext } from "../../common/SplitView/context";
+import { useActiveEditorActions } from "../../editor/active-editor-hooks";
+import { useScrollablePanelAncestor } from "../../workbench/ScrollablePanel";
+import { ToolkitTopic, ToolkitTopicItem } from "./model";
+import MoreButton from "./MoreButton";
+
+interface TopicItemProps extends BoxProps {
+  topic: ToolkitTopic;
+  item: ToolkitTopicItem;
+  detail?: boolean;
+  onForward: () => void;
+}
+
+/**
+ * A toolkit topic item. Can be displayed without detail (for the listing)
+ * or with detail for the "More info" view.
+ *
+ * We show a pop-up over the code on hover to reveal the full code, overlapping
+ * the sidebar scroll area.
+ */
+const TopicItem = ({
+  topic,
+  item,
+  detail,
+  onForward,
+  ...props
+}: TopicItemProps) => {
+  const actions = useActiveEditorActions();
+  const [hovering, setHovering] = useState(false);
+  // Strip the imports.
+  const code = item.code.replace(/^\s*(from[ ]|import[ ]).*$/gm, "").trim();
+  const codeRef = useRef<HTMLDivElement>(null);
+  const lines = code.trim().split("\n").length;
+  const textHeight = lines * 1.5 + "em";
+  const codeHeight = `calc(${textHeight} + var(--chakra-space-5) + var(--chakra-space-5))`;
+  return (
+    <Stack spacing={3} {...props}>
+      <Text as="h3" fontSize="lg" fontWeight="semibold">
+        {item.name}
+      </Text>
+      <Text fontSize="sm">{item.text}</Text>
+      <Box>
+        <Box height={codeHeight}>
+          <Code
+            onMouseEnter={() => setHovering(true)}
+            onMouseLeave={() => setHovering(false)}
+            value={code}
+            position="absolute"
+            ref={codeRef}
+          />
+          {hovering && (
+            <CodePopUp
+              setHovering={setHovering}
+              value={code}
+              codeRef={codeRef}
+            />
+          )}
+        </Box>
+        <HStack spacing={3}>
+          <Button
+            fontWeight="normal"
+            color="white"
+            borderColor="rgb(141, 141, 143)"
```

Comment:
These colours etc. are hardcoded for now - I'll fix this when we have the actual design and I can better appreciate how they fit in with the theme.